test_scripts/Smoke/Resumption/ATF_Resumption_big_amount_of_data.lua

```diff
@@ -0,0 +1,185 @@
+function Test:AddSubMenu()
+  for i = 1, 20 do
+    self.mobileSession:SendRPC("AddSubMenu", { menuID = i, position = 500,
+                menuName = "SubMenu" .. tostring(i)})
+  end
+  local on_hmi_call = EXPECT_HMICALL("UI.AddSubMenu"):Times(20)
+  on_hmi_call:Do(function(_, data)
+    self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  EXPECT_RESPONSE("AddSubMenu", { success = true, resultCode = "SUCCESS" }):Times(20)
+  EXPECT_NOTIFICATION("OnHashChange"):Do(function(_, data)
+    self.currentHashID = data.payload.hashID
+  end):Times(20)
+end
+
+function Test:AddChoiceSet()
+  for i = 1, 20 do
+    self.mobileSession:SendRPC("CreateInteractionChoiceSet", {interactionChoiceSetID = i,
+        choiceSet = { { choiceID = i, menuName = "Choice" .. tostring(i), vrCommands = { "VrChoice" .. tostring(i)}}}})
+  end
+    local on_hmi_call = EXPECT_HMICALL("VR.AddCommand"):Times(20)
+    on_hmi_call:Do(function(_,data)
+      grammar_ID= data.params.grammarID
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    EXPECT_RESPONSE("CreateInteractionChoiceSet", { success = true, resultCode = "SUCCESS" }):Times(20)
+    EXPECT_NOTIFICATION("OnHashChange"):Do(function(_, data)
+      self.currentHashID = data.payload.hashID
+    end):Times(20)
+end
+
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Transport unexpected disconnect. Media app resume at FULL level")
+
+function Test:Close_Session()
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {unexpectedDisconnect = true, 
+    appID = self.applications[default_app_params]})
+  self.mobileSession:Stop()
+end
+
+function Test:Register_And_Resume_App_And_Resumes_Data()
+  local mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+  local on_rpc_service_started = mobileSession:StartRPC()
+  on_rpc_service_started:Do(function()  
+    config.application1.registerAppInterfaceParams.hashID = self.currentHashID
+    Test:expect_Resumption_Data()
+    commonStepsResumption:RegisterApp(default_app_params)   
+    commonStepsResumption:ExpectResumeAppFULL(default_app_params)
+  end)
+end
+
+function Test:expect_Resumption_Data()
+  local on_ui_sub_menu_added = EXPECT_HMICALL("UI.AddSubMenu"):Times(20)
+  on_ui_sub_menu_added:Do(function(_,data)
+    self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS")
+  end)
+  on_ui_sub_menu_added:ValidIf(function(_,data)
+      if data.params.appID == default_app_params.hmi_app_id then
+        return true
+      else 
+        commonFunctions:userPrint(31, "Media app is registered with wrong appID " )
+        return false
+      end
+  end)  
+  local on_vr_commands_added = EXPECT_HMICALL("VR.AddCommand"):Times(40)
+  on_vr_commands_added:Do(function(_,data)
+    self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS")
+  end)
+  on_vr_commands_added:ValidIf(function(_,data)
+    if data.params.type == "Command" or data.params.type == "Choice" then
```

Comment:
@LitvinenkoIra, this verification is not correct. In case SDL has problem and sends 40 type = Command, it will pass, the same if send 40 type = Choice. Will be good also to check that all expected data are received.